lib/root_controller.js

```diff
@@ -0,0 +1,174 @@
+var Tessel = require('./tessel/tessel');
+var logs = require('./logs');
+var debug = require('debug')('controller');
+
+// create an object for t2 root steps for beeing more handy about unit testing of the T2 root command itself.
+var controller = {};
+controller.ssh = {
+  makeMenu: function(tessel, index, rtm) {
+    if (tessel.connection.connectionType === 'LAN') {
+      if (!tessel.connection.authorized) {
+        rtm.add(index + ') ' + tessel.name + ': ' + tessel.connection.ip + ' (not authorized) \n');
+      } else {
+        rtm.add(index + ') ' + tessel.name + ': ' + tessel.connection.ip + ' (authorized) \n');
+      }
+
+    } else if (tessel.connection.connectionType === 'USB') {
+      if (!tessel.connection.authorized) {
+        rtm.add(index + ') ' + tessel.name + ': [USB] (not authorized) \n');
+      } else {
+        rtm.add(index + ') ' + tessel.name + ': [USB] (authorized) \n');
+      }
+    } else {
+      rtm.add('EXIT');
+    }
+    // FIXME: Workaround for trouble with dynamic added menu elements
+    rtm._draw();
+  },
+  runSSH: function(id, opts, tessels, resolve, reject) {
+    // clear console
+    debug('runSSH...\n root@' + tessels[id].connections[0].host);
+    if (opts.menu) {
+      // because the test needs to resolve the promise ...
+      resolve({
+        opts: opts,
+        tessels: tessels
+      });
+    } else {
+      // if this is no test, starting the menu as child process
+      process.stdout.write('\u001b[2J\u001b[0;0H');
+      var ch = require('child_process')
+        .spawn('ssh', ['-i',
+          opts.path,
+          'root@' + tessels[id].connections[0].ip
+        ], {
+          stdio: 'inherit'
+        });
+      logs.info('Connect to ' + tessels[id].connections[0].host + '...');
+
+      // FIXME: There is no handler for poweroff Tessel ! Needs to be fixed within the firmware...
+      // (Terminal freezes imediatelly after poweroff the Tessel)
+
+      ch.once('error', function(e) {
+        if (e === 255) {
+          process.stdout.write('\u001b[2J\u001b[0;0H');
+          logs.warn('Sorry, you are not authorized!');
+          logs.info('"t2 key generate" might help :-)');
```

Comment:
@Frijol hmm... the `t2 root` implementation currently seek for Tessels before attempting to login. While doing this it doesn't do heuristics and filtering of non authorized Tessels. The reason is, there are many cases the list of Tessels will variegate. Also there is not global List of Tessels I could attempt. One more reason is the use of option `t2 root -i ~\.ssh\id_rsa` what will fail on a not authorized Tessel. In that case the child_process will error with code 255. Another case would be, a user is accidental switched to superuser mode and uses `t2 root` what would use `/root/.tessel/id_rsa` or no key.

hmm ... maybe the wording should address this used case also (any suggestions)?

In general, the new branch for pull requesting (not published now) provides a method because this phrase is used often.
The call looks something like this:

```
controller.ssh.notAuthorized();
```
I have done this in the current working branch `menu` also. Here https://github.com/Student007/t2-cli/blob/menu/lib/controller.js#L405-L408 you can see the suggested divided objects `controller.ssh` and `controller.menu`